index.js

```diff
@@ -536,31 +580,11 @@ class K8sExecutor extends Executor {
                 return this.breaker.runCommand(statusOptions);
             })
             .then(res => {
-                const waitingReason = hoek.reach(res.body, CONTAINER_WAITING_REASON_PATH);
-                const status = res.body.status.phase.toLowerCase();
-
-                if (status === 'failed' || status === 'unknown') {
-                    throw new Error(`Failed to create pod. Pod status is:${JSON.stringify(res.body.status, null, 2)}`);
-                }
-
-                if (
-                    waitingReason === 'CrashLoopBackOff' ||
-                    waitingReason === 'CreateContainerConfigError' ||
-                    waitingReason === 'CreateContainerError' ||
-                    waitingReason === 'StartError'
-                ) {
-                    throw new Error('Build failed to start. Please reach out to your cluster admin for help.');
-                }
-
-                if (
-                    waitingReason === 'ErrImagePull' ||
-                    waitingReason === 'ImagePullBackOff' ||
-                    waitingReason === 'InvalidImageName'
-                ) {
-                    throw new Error('Build failed to start. Please check if your image is valid.');
-                }
-
-                return null;
+                return this.checkPodResponse(res)
```

Comment:
this can be just `.this(res => this.checkPodResponse(res))`

Reply:
u mean - .then(res => this.checkPodResponse(res)) ?